core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -2599,6 +2602,22 @@ default boolean allowedInOr(RelOptPredicateList predicates) {
     }
   }
 
+  /**
+   * Visitor which finds all inputs used by an expressions.
+   */
+  private static class VariableCollector extends RexVisitorImpl<Void> {
+    private final Set<RexInputRef> refs = new LinkedHashSet<>();
```

Comment:
Just curious, why specifically a `LinkedHashSet` here?

Reply:
I have just copied it from other place and didn't reflect enough on reasoning. Can be simplified with HashMap.

Reply:
I don't think it will make much of a difference, so it's fine as-is for me

Reply:
I have the same feeling with @asolimando, usually a `LinkedHashSet` would make me wonder what's its purpose to use it instead of a `HashSet`, which will increase the cost of understanding.

If we have a reason, we'd better write it into the comment. Else, I would prefer to use the most general one for the above reason. WDYT?

Reply:
Agree, fixed.